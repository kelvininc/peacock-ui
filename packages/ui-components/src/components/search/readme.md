# kv-search



<!-- Auto Generated Below -->


## Usage

### Angular

```html
<!-- Default -->
<kv-search></kv-search>

<!-- Different Placeholder -->
<kv-search placeholder="New Search Placeholder"></kv-search>

<!-- Disabled -->
<kv-search disabled></kv-search>
```


### React

```tsx
import React from 'react';

import { KvSearch } from '@kelvininc/react-ui-components';

export const SearchExample: React.FC = () => (
	<>
		{/*-- Default --*/}
		<KvSearch />

		{/*-- With Placeholder --*/}
		<KvSearch placeholder="New Search Placeholder" />

		{/*-- Disabled --*/}
		<KvSearch disabled />
	</>
);

```



## Properties

| Property      | Attribute     | Description                                                        | Type                                           | Default                |
| ------------- | ------------- | ------------------------------------------------------------------ | ---------------------------------------------- | ---------------------- |
| `disabled`    | `disabled`    | (optional) Search disabled                                         | `boolean`                                      | `false`                |
| `placeholder` | `placeholder` | (optional) Search place holder                                     | `string`                                       | `'Search'`             |
| `size`        | `size`        | (optional) Sets this tab item to a different styling configuration | `EComponentSize.Large \| EComponentSize.Small` | `EComponentSize.Large` |
| `value`       | `value`       | Search value                                                       | `string`                                       | `undefined`            |


## Events

| Event              | Description                              | Type                      |
| ------------------ | ---------------------------------------- | ------------------------- |
| `clickResetButton` | Emitted when the reset buccon is clicked | `CustomEvent<MouseEvent>` |
| `textChange`       | Emitted when a keyboard input occurred   | `CustomEvent<string>`     |
| `textFieldBlur`    | Emitted when text field lost focus       | `CustomEvent<string>`     |


## Dependencies

### Used by

 - [kv-select](../select)

### Depends on

- [kv-text-field](../text-field)
- [kv-icon](../icon)

### Graph
```mermaid
graph TD;
  kv-search --> kv-text-field
  kv-search --> kv-icon
  kv-text-field --> kv-tooltip
  kv-text-field --> kv-form-label
  kv-text-field --> kv-icon
  kv-text-field --> kv-form-help-text
  kv-form-help-text --> kv-icon
  kv-select --> kv-search
  style kv-search fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


