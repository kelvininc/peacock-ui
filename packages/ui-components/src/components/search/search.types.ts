import { EventEmitter } from '@stencil/core';

export interface ISearchEvents {
	/** Emitted when the reset buccon is clicked */
	clickResetButton: EventEmitter<MouseEvent>;
}
