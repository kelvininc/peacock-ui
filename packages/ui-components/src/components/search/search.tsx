import { Component, h, Event, EventEmitter, Prop, Host } from '@stencil/core';
import { isEmpty } from 'lodash-es';
import { EComponentSize, ITooltip } from '../../types';
import { EIconName } from '../icon/icon.types';
import { EInputFieldType, EValidationState } from '../text-field/text-field.types';
import { ISearch, ISearchEvents } from './search.types';

@Component({
	tag: 'kv-search',
	shadow: true
})
export class KvSearch implements ISearch, ISearchEvents {
	/** @inheritdoc */
	@Prop({ reflect: true }) type: EInputFieldType = EInputFieldType.Text;
	/** @inheritdoc */
	@Prop({ reflect: true }) label?: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) examples?: string[];
	/** @inheritdoc */
	@Prop({ reflect: true }) inputName?: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) placeholder?: string = 'Search';
	/** @inheritdoc */
	@Prop({ reflect: true }) maxLength?: number;
	/** @inheritdoc */
	@Prop({ reflect: true }) minLength?: number;
	/** @inheritdoc */
	@Prop({ reflect: true }) max?: string | number;
	/** @inheritdoc */
	@Prop({ reflect: true }) min?: string | number;
	/** @inheritdoc */
	@Prop({ reflect: true }) step?: string | number;
	/** @inheritdoc */
	@Prop() size: EComponentSize = EComponentSize.Large;
	/** @inheritdoc */
	@Prop({ reflect: true }) disabled: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) required: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) loading: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) state: EValidationState = EValidationState.None;
	/** @inheritdoc */
	@Prop({ reflect: true }) readonly: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) helpText: string | string[] = [];
	/** @inheritdoc */
	@Prop({ reflect: true }) forcedFocus: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) forcedInputFocus: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) highlighted: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) tooltipConfig?: Partial<ITooltip>;
	/** @inheritdoc */
	@Prop({ reflect: true }) value?: string | number | null = '';
	/** @inheritdoc */
	@Prop({ reflect: true }) useInputMask?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) inputMaskRegex?: string = '';

	/** @inheritdoc */
	@Event() textChange: EventEmitter<string>;
	/** @inheritdoc */
	@Event() textFieldBlur: EventEmitter<string>;
	/** @inheritdoc */
	@Event() clickResetButton: EventEmitter<MouseEvent>;
	/** @inheritdoc */
	@Event() rightActionClick: EventEmitter<MouseEvent>;

	private onTextChange = (event: CustomEvent<string>) => {
		event.stopPropagation();
		this.textChange.emit(event.detail);
	};

	private onResetClick = (event: CustomEvent<MouseEvent>) => {
		event.stopPropagation();
		this.clickResetButton.emit(event.detail);
		this.textChange.emit();
	};

	render() {
		const shouldShowResetIcon = !isEmpty(this.value) && !this.disabled;
		return (
			<Host>
				<kv-text-field
					actionIcon={shouldShowResetIcon ? EIconName.Close : undefined}
					onTextChange={this.onTextChange}
					onRightActionClick={this.onResetClick}
					type={this.type}
					label={this.label}
					examples={this.examples}
					icon={EIconName.Search}
					inputName={this.inputName}
					placeholder={this.placeholder}
					maxLength={this.maxLength}
					minLength={this.minLength}
					max={this.max}
					min={this.min}
					step={this.step}
					size={this.size}
					disabled={this.disabled}
					required={this.required}
					loading={this.loading}
					state={this.state}
					readonly={this.readonly}
					helpText={this.helpText}
					forcedFocus={this.forcedFocus}
					forcedInputFocus={this.forcedInputFocus}
					highlighted={this.highlighted}
					tooltipConfig={this.tooltipConfig}
					value={this.value}
					useInputMask={this.useInputMask}
					inputMaskRegex={this.inputMaskRegex}
				/>
			</Host>
		);
	}
}
