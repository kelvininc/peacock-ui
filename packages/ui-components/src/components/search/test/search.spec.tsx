import { newSpecPage } from '@stencil/core/testing';
import { KvSearch } from '../search';

describe('search', () => {
	it('renders', async () => {
		const page = await newSpecPage({
			components: [KvSearch],
			html: `<kv-search></kv-search>`,
		});
		expect(page.root).toEqualHtml(`
			<kv-search>
				<mock:shadow-root>
				<slot></slot>
				</mock:shadow-root>
			</kv-search>
		`);
	});
});
