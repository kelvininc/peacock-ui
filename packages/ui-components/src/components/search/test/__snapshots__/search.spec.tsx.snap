// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search (unit tests) when is disabled should match the snapshot 1`] = `
<kv-search disabled="" placeholder="Search">
  <mock:shadow-root>
    <kv-text-field disabled="" icon="kv-search" placeholder="Search" size="large"></kv-text-field>
  </mock:shadow-root>
</kv-search>
`;

exports[`Search (unit tests) when uses default props should match the snapshot 1`] = `
<kv-search placeholder="Search">
  <mock:shadow-root>
    <kv-text-field icon="kv-search" placeholder="Search" size="large"></kv-text-field>
  </mock:shadow-root>
</kv-search>
`;

exports[`Search (unit tests) when value is changed should match the snapshot 1`] = `
<kv-search placeholder="Search" value="Teste">
  <mock:shadow-root>
    <kv-text-field actionicon="kv-close" icon="kv-search" placeholder="Search" size="large" value="Teste"></kv-text-field>
  </mock:shadow-root>
</kv-search>
`;
